CHANGELOG.md

```diff
@@ -2,6 +2,86 @@
+block addition was fixed. Also, those who use NeoGo node with
+EnableCORSWorkaround setting may want to update to be able to handle web-socket
+requests with Origin set in header. Finally, we're kindly asking our users to
+make acquaintance with our roadmap file where deprecated code removal schedule
+is placed.
+
+This version is fully compatible with C# node 3.4.0 and does not require
+resynchronization on upgrade. This time for sure, the next release is planned
+to be compatible with 3.5.0.
+
+New features:
+ * logarithmic gossip fan-out, new `BroadcastFactor` application setting and
+   adaptive peer requests are added to optimise broadcasting process (#608,
+   #2678, #2743)
+ * Prometheus histograms with handling time are added for P2P commands (#2750)
+   and RPC calls (#2784)
+ * transaction awaiting is added to RPC clients (#2740, #2749)
+ * RPC bindings generator (#2705) is added for both safe (#2766) and
+   state-changing (#2769, #2778) methods with initial iterator support (#2783)
+ * the ability to reset chain state to a particular height is added (#2576,
+   #2773)
+
+Behaviour changes:
+ * network packets sending is now limited in time (#2738)
+ * broadcast messages will be sent concurrently to each peer (#2741)
+ * ping messages are broadcast now (#2742)
+ * transactions requested by dBFT will be accepted on demand only (#2746, #2748,
+   nspcc-dev/dbft#63)
+ * Aspidochelone fork block for NeoFS sidechain mainnet configuration is
+   rescheduled (#2754, #2776)
+ * `NOT` opcode for Bool emission is now used to reduce resulting script
+   cost (#2762)
+ * RPC server will now handle any Origin in websocket connection when
+   EnableCORSWorkaround setting is on (#2772, #2724)
+
+Improvements:
+ * network code refactoring (#2742)
+ * VM CLI's state-based functionality improvements (extended signers and arguments
+   parsing, enabled logging) and additional commands (`loadtx`, `loaddeployed`,
+   `jump`) (#2729, #2606, #2740)
+ * incoming P2P transactions are now processed concurrently and buffer for
+   inventory hashes is reusable (#2755)
+ * investigation of the network I/O timeouts during bench test was made (#2744) // TODO: do we need to put this issue here? We have all the related PRs mentioned already.
+ * `GetData` and `GetBlocksByIndex` replies are now batched on a TCP-level (#2757)
+ * incoming P2P transactions are checked against currently processing ones (#2758)
+ * incoming P2P blocks check optimisation for recently added blocks is added (#2759)
+ * batch size of broadcast transactions hashes is adjusted (#2760)
+ * improve peer reconnections logic and send `GetAddr` messages more often
+   (#2761, #2745)
+ * clarify documentation for wallet configuration file (#2763)
+ * WS RPC client subscriptions API is extended to accept custom subscriber
+   channels (#2756, #2764)
+ * security vulnerability of Koblitz curve implementation was fixed with dropped
```

Comment:
I think there was no vulnerability in our case.

Reply:
You're right, it sounds like we had it, so I've rephrased the sentence.